etl/scripts-ccao-data-warehouse-us-east-1/spatial/spatial-ccao-corner.R

```diff
@@ -0,0 +1,236 @@
+
+# Iterate over the years and townships, saving the results to a temporary file
+# after each iteration
+for (iter_year in parcel_years) {
+  tictoc::tic(paste("Finished processing corners for:", iter_year))
+
+  # Load the full year's parcel file to iterate though by township
+  parcels <- open_dataset(parcel_path) %>%
+    filter(year == iter_year) %>%
+    geoarrow_collect_sf()
+
+  for (iter_town in ccao::town_dict$township_code) {
+    print(paste("Now processing township:", iter_town))
+
+    remote_file <- file.path(
+      output_bucket,
+      glue("year={iter_year}"),
+      glue("township_code={iter_town}"),
+      "part-0.parquet"
+    )
+
+    town_parcels <- parcels %>%
+      filter(town_code == iter_town)
+
+    # Get a bounding box 3,000 feet larger than the extent of the town parcels.
+    # Used to query OSM for streets and to ensure parcels on the "edge" of the
+    # township have their correct set of neighbors to check in Step 3 below
+    town_bbox <- town_parcels %>%
+      st_set_geometry("geometry_3435") %>%
+      st_bbox() %>%
+      st_as_sfc() %>%
+      st_buffer(3000, endCapStyle = "FLAT", joinStyle = "MITRE") %>%
+      st_transform(4326) %>%
+      st_bbox()
+
+    # Get a buffered copy of the town parcels. This is used in Step 3 below in
+    # order to fill any tiny gaps between abutting parcels
+    town_parcels_buf <- parcels %>%
+      filter(
+        between(lon, town_bbox$xmin, town_bbox$xmax) &
+          between(lat, town_bbox$ymin, town_bbox$ymax)
+      ) %>%
+      st_buffer(dist = units::set_units(2, "m"))
+
+    # Fetch the OSM street network for the township, removing any OSM way types
+    # that are not main roads
+    osm_streets <- opq(bbox = town_bbox) %>%
+      add_osm_feature(key = "highway") %>%
+      osmdata_sf() %>%
+      .$osm_lines %>%
+      filter(
+        !highway %in% c(
+          "bridleway", "construction", "corridor", "cycleway", "elevator",
+          "service", "services", "steps", "platform", "motorway",
+          "motorway_link", "pedestrian", "track", "path", "footway", "alley"
+        )
+      ) %>%
+      st_transform(3435)
+
+    # Step 1: Find the minimum rectangle that bounds the parcel, then use that
+    # rectangle to determine the parcel's orientation and length. These values
+    # are used in the next step to draw a cross on the parcel
+    town_mrr <- town_parcels %>%
+      st_minimum_rotated_rectangle() %>%
+      select(lon, lat, geometry = geometry_3435) %>%
+      mutate(id = row_number()) %>%
+      st_transform(4326) %>%
+      st_cast("POINT") %>%
+      mutate(
+        length = st_distance(geometry, lead(geometry), by_element = TRUE) +
+          units::set_units(40, "m"),
+        bearing = c(
+          lwgeom::st_geod_azimuth(geometry),
+          units::set_units(NA, "radians")
+        ),
+        bearing = units::set_units(bearing, "degrees")
+      ) %>%
+      filter(row_number() %% 5 != 0) %>%
```

Comment:
**[Question, non-blocking]** What's going on here?

Reply:
Added a clarifying comment in 6c7a4c4f!